> What is the problem with windows dynamic?
> 
> ATM it installs man1 manual pages (i.e. for tools) but it doesn't install a tool (feature). (They are the "examples".)

It fails with the following linker error:

```
[12/13] cmd.exe /C "cd . && D:\vcpkg\downloads\tools\cmake-3.21.1-windows\cmake-3.21.1-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=CMakeFiles\utils.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x86\link.exe  CMakeFiles\utils.dir\examples\utils.cc.obj  /out:utils.dll /implib:utils.lib /pdb:utils.pdb /dll /version:0.0 /machine:X86 /nologo    /debug /INCREMENTAL  D:\vcpkg\installed\x86-windows\debug\lib\libpng16d.lib  D:\vcpkg\installed\x86-windows\debug\lib\zlibd.lib  D:\vcpkg\installed\x86-windows\debug\lib\jpeg.lib  opengl32.lib  D:\vcpkg\installed\x86-windows\debug\lib\freeglutd.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
FAILED: utils.dll utils.lib 
cmd.exe /C "cd . && D:\vcpkg\downloads\tools\cmake-3.21.1-windows\cmake-3.21.1-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=CMakeFiles\utils.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x86\link.exe  CMakeFiles\utils.dir\examples\utils.cc.obj  /out:utils.dll /implib:utils.lib /pdb:utils.pdb /dll /version:0.0 /machine:X86 /nologo    /debug /INCREMENTAL  D:\vcpkg\installed\x86-windows\debug\lib\libpng16d.lib  D:\vcpkg\installed\x86-windows\debug\lib\zlibd.lib  D:\vcpkg\installed\x86-windows\debug\lib\jpeg.lib  opengl32.lib  D:\vcpkg\installed\x86-windows\debug\lib\freeglutd.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
LINK Pass 1: command "C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x86\link.exe CMakeFiles\utils.dir\examples\utils.cc.obj /out:utils.dll /implib:utils.lib /pdb:utils.pdb /dll /version:0.0 /machine:X86 /nologo /debug /INCREMENTAL D:\vcpkg\installed\x86-windows\debug\lib\libpng16d.lib D:\vcpkg\installed\x86-windows\debug\lib\zlibd.lib D:\vcpkg\installed\x86-windows\debug\lib\jpeg.lib opengl32.lib D:\vcpkg\installed\x86-windows\debug\lib\freeglutd.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\utils.dir/intermediate.manifest CMakeFiles\utils.dir/manifest.res" failed (exit code 1120) with the following output:
utils.cc.obj : error LNK2019: unresolved external symbol "public: void __thiscall sjpeg::EncoderParam::ResetMetadata(void)" (?ResetMetadata@EncoderParam@sjpeg@@QAEXXZ) referenced in function "class std::vector<unsigned char,class std::allocator<unsigned char> > __cdecl ReadJPEG(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int * const,int * const,struct sjpeg::EncoderParam * const)" (?ReadJPEG@@YA?AV?$vector@EV?$allocator@E@std@@@std@@ABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@2@QAH1QAUEncoderParam@sjpeg@@@Z)
```

I did not further investigate this error since the library is missing dllexport annotations and a DEF file, so it is obviously not intended to be built as a dynamic library on Windows.

--- a/ports/sjpeg/portfile.cmake
+++ b/ports/sjpeg/portfile.cmake
@@ -21,5 +21,6 @@ vcpkg_cmake_config_fixup(CONFIG_PATH share/sjpeg/cmake)
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share/man")
 
 file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/versions/s-/sjpeg.json
+++ b/versions/s-/sjpeg.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "229da3d668d19eedde8478ae0aa5bf3e9c18eeb0",
+      "git-tree": "fe20a97b3cdd0a6ad84ca9543670ea61673d1026",
       "version-date": "2021-10-31",
       "port-version": 0
     }

